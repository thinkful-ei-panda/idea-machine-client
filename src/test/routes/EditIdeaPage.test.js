import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import EditIdeaPage from '../../routes/EditIdeaPage';
describe(`EditIdeaPage Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <EditIdeaPage />
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <EditIdeaPage />
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})