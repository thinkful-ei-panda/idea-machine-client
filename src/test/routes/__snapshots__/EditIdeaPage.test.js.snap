// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditIdeaPage Component Snapshot test Renders the UI as expected 1`] = `
<div
  className="AddIdeaPage"
>
  <h2>
    Edit Idea
  </h2>
  <div>
    <form
      onSubmit={[Function]}
    >
      <legend>
        Edit Idea Form
      </legend>
      <fieldset>
        <div
          className="column"
        >
          <div
            className="formInputContainer"
          >
            <label
              htmlFor="title"
            >
              Title
            </label>
            <input
              id="title"
              name="title"
            />
          </div>
          <div
            className="formInputContainer"
          >
            <label
              htmlFor="content"
            >
              Content
            </label>
            <textarea
              id="content"
              name="content"
            />
          </div>
          <div
            className="buttonContainer"
          >
            <button>
              Finish Edit
            </button>
          </div>
        </div>
      </fieldset>
    </form>
  </div>
</div>
`;
